pkg/asset/manifests/azure/cluster.go

```diff
@@ -104,6 +104,30 @@ func GenerateClusterAssets(installConfig *installconfig.InstallConfig, clusterID
 	}
 
 	virtualNetworkID := ""
+	lbip := capz.DefaultInternalLBIPAddress
+	machineCidr := installConfig.Config.MachineNetwork
+	// Check if default lbip is within machine network.
+	isInMachineCIDR := false
+	for _, cidrRange := range machineCidr {
+		if cidrRange.CIDR.Contains(net.ParseIP(lbip)) {
+			isInMachineCIDR = true
+			break
+		}
+	}
+
+	// If not in machine network, assign the first IP in the CIDR to lbip.
+	if !isInMachineCIDR {
+		ip := machineCidr[0].CIDR.IP
+		// Since the first IP of the subnets are usually reserved, pick the next one that's available in the CIDR.
+		ip[len(ip)-1]++
+		lbip = ip.String()
```

Comment:
According to the doc jinyun linked to it's the first three ips that are reserved:

> Azure assigns private IP addresses to resources from the address range of the virtual network subnet where the resource is.
> 
> Azure reserves the first four addresses in each subnet address range. The addresses can't be assigned to resources. For example, if the subnet's address range is 10.0.0.0/16, addresses 10.0.0.0-10.0.0.3 and 10.0.255.255 are unavailable. IP addresses within the subnet's address range can only be assigned to one resource at a time.

So it seems  you need to pick at least the 4th ip address?

Reply:
I feel like +4 has a higher chance to fail but let me do it anyway